openslides/core/chatbox.py

```diff
@@ -20,14 +21,13 @@ def on_open(self, info):
         """
         from openslides.participant.models import User
 
-        # TODO: Use the django way to get the session to be compatible with
-        # other auth-backends; see comment in pull request #1220:
-        # https://github.com/OpenSlides/OpenSlides/pull/1220#discussion_r11565705
-        session_key = info.get_cookie(settings.SESSION_COOKIE_NAME).value
-        session = Session.objects.get(session_key=session_key)
+        # get the session (compatible with other auth-backends)
+        engine = import_module(settings.SESSION_ENGINE)
         try:
+            session_key = info.get_cookie(settings.SESSION_COOKIE_NAME).value
+            session = engine.SessionStore(session_key)
             self.user = User.objects.get(pk=session.get_decoded().get('_auth_user_id'))
-        except User.DoesNotExist:
+        except (User.DoesNotExist, AttributeError):
```

Comment:
It is already reduced to the relevant lines of code:
An AttributeError could be arise in line 27 (.value) and 29 (.get_decoded()).
An User.DoesNotExist error could be arise in line 29 (User.objects.get..)
So what is to optimize here?